pkg/core/tekton/builder/task_spec.go

```diff
@@ -0,0 +1,71 @@
+package builder
+
+import (
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// TaskSpecBuilder holds the tekton task spec definition.
+type TaskSpecBuilder struct {
+	TaskSpec v1beta1.TaskSpec
+}
+
+// NewTaskSpecBuilder creates a TaskSpec with default values.
+func NewTaskSpecBuilder(name, image, command string) *TaskSpecBuilder {
+	b := &TaskSpecBuilder{}
+	b.TaskSpec = v1beta1.TaskSpec{
+		Steps: []v1beta1.Step{{Container: corev1.Container{
```

Comment:
Does this mean that all the tasks you generate will only have one step ?

Reply:
Yes, currently there is no way to map a FuseML workflow step to multiple steps in a Tekton Task.

Reply:
okie dokie. For runnables, I suspect that a runnable will be represented in a pipeline by as many as 3 steps in a single task: one that prepares the inputs, one that is actually represented by the runnable, and a 3rd one to prepare the outputs. The fact that steps in a task are implemented by containers running in the same pod and having access to the same set of volumes seems to support this idea. But we'll revisit this later.